crates/rspack_core/src/compiler/hmr.rs

```diff
@@ -424,6 +424,7 @@ where
                 .map(|hash| hash.rendered(self.compilation.options.output.hash_digest_length)),
             ),
           );
+          asset.info.version = Default::default();
```

Comment:
Is this needed? We can do it in constructor

Reply:
`CompilationAsset` constructor uses entry.info directly on line 411, it seems strange to reset the version in `CompilationAsset` constructor. Maybe it's better to reset it like `with_hot_module_replacement`